src/test/java/com/iexec/sms/secret/SecretControllerTests.java

```diff
@@ -0,0 +1,318 @@
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.of(new Web3Secret()));
+        assertThat(secretController.addWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, WEB3_SECRET_VALUE))
+                .isEqualTo(ResponseEntity.status(HttpStatus.CONFLICT).build());
+    }
+
+    @Test
+    void addWeb3Secret() {
+        when(authorizationService.getChallengeForSetWeb3Secret(WEB3_SECRET_ADDRESS, WEB3_SECRET_VALUE))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByOwner(CHALLENGE, AUTHORIZATION, WEB3_SECRET_ADDRESS))
+                .thenReturn(true);
+        when(web3SecretService.getSecret(WEB3_SECRET_ADDRESS))
+                .thenReturn(Optional.empty());
+        assertThat(secretController.addWeb3Secret(AUTHORIZATION, WEB3_SECRET_ADDRESS, WEB3_SECRET_VALUE))
+                .isEqualTo(ResponseEntity.noContent().build());
+    }
+    //endregion
+
+    //region isWeb2SecretSet
+    @Test
+    void shouldReturnNoContentWhenWeb2SecretExists() {
+        when(web2SecretsService.getSecret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .thenReturn(Optional.of(new Secret()));
+        assertThat(secretController.isWeb2SecretSet(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .isEqualTo(ResponseEntity.noContent().build());
+        verifyNoInteractions(authorizationService, web3SecretService);
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenWeb2SecretDoesNotExist() {
+        when(web2SecretsService.getSecret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .thenReturn(Optional.empty());
+        assertThat(secretController.isWeb2SecretSet(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .isEqualTo(ResponseEntity.notFound().build());
+        verifyNoInteractions(authorizationService, web3SecretService);
+    }
+    //endregion
+
+    //region getWeb2Secret
+    @Test
+    void failToGetWeb2SecretWhenBadAuthorization() {
+        when(authorizationService.getChallengeForGetWeb2Secret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByHimself(CHALLENGE, AUTHORIZATION, WEB2_OWNER_ADDRESS))
+                .thenReturn(false);
+        assertThat(secretController.getWeb2Secret(AUTHORIZATION, WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME, false))
+                .isEqualTo(ResponseEntity.status(HttpStatus.UNAUTHORIZED).build());
+        verifyNoInteractions(web2SecretsService, web3SecretService);
+    }
+
+    @Test
+    void failToGetWeb2SecretWhenSecretDoesNotExist() {
+        when(authorizationService.getChallengeForGetWeb2Secret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME))
+                .thenReturn(CHALLENGE);
+        when(authorizationService.isSignedByHimself(CHALLENGE, AUTHORIZATION, WEB2_OWNER_ADDRESS))
+                .thenReturn(true);
+        when(web2SecretsService.getSecret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME)).thenReturn(Optional.empty());
```

Comment:
```suggestion
        when(web2SecretsService.getSecret(WEB2_OWNER_ADDRESS, WEB2_SECRET_NAME, false)).thenReturn(Optional.empty());
```
Same as before, I guess.

Reply:
Fixed in 5a1d5f6